templates/haproxy.cfg

```diff
@@ -52,9 +52,19 @@ frontend fe_http
 
 frontend fe_https
     bind *:443 ssl crt {{ LOAD_BALANCER_CERTS_DIR }}
+    acl fail_mode status 500
+    acl fail_mode status 502
+    acl fail_mode status 504
```

Comment:
Can you please reorder the statements here?  I would prefer to have ACLs matching the request and `http-request` statements first, then the calls to the backends, and finally the processing of the response (ACLs for status code, `http-response` statements), since this matches the order in which things are actually processed.  (This is just a nit to make this more readable – it shouldn't change the behaviour of the config in any way.)

Reply:
@smarnach  I left out 503 because that causes a redirect loop because both ``errorloc 503`` and the ``acl`` cause it to redirect. I did not expect this but that is the result I got during local testing. 

I will use look up the range syntax and use that if possible (skipping 503).

Reply:
Sorry I was wrong about why it causes a loop. The loop is because the maintenance page itself needs to raise an error, which is 503. If we redirect on that error then it will perform an infinite redirect. I had used 503 in the maintenance page because it seems apt and is also not used by the LMS.